### PR TITLE
CRUD for institutions

### PR DESCRIPTION
**Pull request by Troi**

### CODE REVIEW

File: src/BeneficiaryBundle/Repository/InstitutionRepository.php [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    }
+
+    /**
+     * Create sub request to get institutions in country.
+     * The institution address location must be in the country ($countryISO3).
+     *
+     * @param QueryBuilder $qb
+     * @param $countryISO3
+     */
+    public function whereInstitutionInCountry(QueryBuilder &$qb, $countryISO3)
```
Comment (iarro) [inline src/BeneficiaryBundle/Repository/InstitutionRepository.php:206]: IMHO its not correct approach. Why `whereInstitutionInCountry()` is public?

Comment (Troi) [reply src/BeneficiaryBundle/Repository/InstitutionRepository.php]: it is relict I should deleted before commit

File: src/BeneficiaryBundle/Controller/InstitutionController.php [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+     */
+    public function createAction(Request $request)
+    {
+        $requestArray = $request->request->all();
+
+        $requestRequirements = new OptionsResolver();
+        $requestRequirements->setRequired('institution');
+        $requestRequirements->setAllowedTypes('institution', 'array');
+        $requestRequirements->setDefaults([
+            '__country' => 'KHM',
```
[the remark this answers is not in the document]
Comment (Troi) [reply src/BeneficiaryBundle/Controller/InstitutionController.php]: ok

File: src/BeneficiaryBundle/Controller/InstitutionController.php [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+        $dataOptionRequirements->setAllowedTypes('sort', 'array');
+        $dataOptionRequirements->setAllowedTypes('pageIndex', 'int');
+        $dataOptionRequirements->setAllowedTypes('pageSize', 'int');
+        $dataOptionRequirements->setAllowedTypes('__country', 'string');
+        $dataOptionRequirements->setDefaults([
+            'filter' => [],
+            'sort' => [],
+            'pageIndex' => 0,
+            'pageSize' => 10,
+            '__country' => 'KHM',
```
[the remark this answers is not in the document]
Comment (Troi) [reply src/BeneficiaryBundle/Controller/InstitutionController.php:96]: ok

**Pull request merged**